src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -166,66 +168,64 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
                 }
             }
         }
-        catch (ProduceException<string, byte[]> e)
+        catch (KafkaException e)
         {
-            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to produce message.{Environment.NewLine}");
-            
-            if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)
-                errorMessageBuilder.Append($"Add an required ACL permissions to your API Key that will make WRITE operations on the '{currentTopic?.Name}' topic possible.");
-            
-            else if (TryGetMissingAclsBindings(out var detailsMessage, currentTopic?.Name))
-                errorMessageBuilder.Append(detailsMessage);
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to produce message. Error code: '{e.Error.Code}'. Error reason: '{e.Error.Reason}'. {Environment.NewLine}");
 
+            if (e is ProduceException<string, byte[]> produceException)
+            {
+                if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)
+                    errorMessageBuilder.Append($"Add an required ACL permissions to your API Key that will make WRITE operations on the '{produceException.DeliveryResult.Topic}' topic possible.");
+                
+                if (TryGetMissingAclsBindings(out var message))
```

Comment:
Shouldn't it be `else if`? If we know that it was `ErrorCode.TopicAuthorizationFailed)` then we already have a clear error message.

Reply:
done https://github.com/ravendb/ravendb/pull/17577/commits/640d4f4056e6e9d834f0db6d03b65f619d19c2ad